table/block_based/block_based_table_reader.cc

```diff
@@ -1182,11 +1187,61 @@ Status BlockBasedTable::GetDataBlockFromCache(
   Status s;
   BlockContents* compressed_block = nullptr;
   Cache::Handle* block_cache_compressed_handle = nullptr;
+  Cache::CreateCallback create_cb;
+  Cache::CacheItemHelperCallback helper_cb;
+  if (std::is_same<TBlocklike, BlockContents>::value) {
```

Comment:
An alternative approach could be to add  ```CacheItemHelperCallback``` to ```BlocklikeTraits```, and have a common templatized implementation of ```CreateCallback``` that calls ```BlocklikeTraits::Create<TBlocklike>```.  It seems cleaner to have all cache related functionality for a block type in one place like ```BlocklikeTraits```. What do you think?

Reply:
Sure, it would be easier to extend in the future. Otherwise, people may forget to add the std::is_same<TBlocklike, "">::value branch to the two places, or even more places in the future.